File: src/main/java/com/github/morningzeng/toolset/ui/coding/URLComponent.java
package com.github.morningzeng.toolset.ui.coding;

import com.intellij.openapi.project.Project;

import java.net.URLDecoder;
import java.net.URLEncoder;
import java.util.function.Supplier;

import static java.nio.charset.StandardCharsets.UTF_8;

/**
 * @author Morning Zeng
 * @since 2024-05-21
 */
public final class URLComponent extends AbstractCodingComponent {

    public URLComponent(final Project project) {
        super(project);
        this.encodeArea.setPlaceholder("URL encoded text");
        this.decodeArea.setPlaceholder("URL decoded text");
    }

    @Override
    Supplier<String> encode() {
        return () -> URLEncoder.encode(this.decodeArea.getText(), UTF_8);
    }

    @Override
    Supplier<String> decode() {
        return () -> URLDecoder.decode(this.encodeArea.getText(), UTF_8);
    }

}

File: src/main/kotlin/com/github/morningzeng/toolbox/ui/view/coding/URLComponent.kt
package com.github.morningzeng.toolbox.ui.view.coding

import com.intellij.openapi.project.Project
import java.net.URLDecoder
import java.net.URLEncoder
import java.nio.charset.StandardCharsets

/**
 * @author Morning Zeng
 * @since 2025-07-22
 */
class URLComponent(
    project: Project
) : AbstractCodingComponent(project) {
    override fun encode(): String {
        return URLEncoder.encode(decodeArea.text, StandardCharsets.UTF_8)
    }

    override fun decode(): String {
        return URLDecoder.decode(encodeArea.text, StandardCharsets.UTF_8)
    }
}